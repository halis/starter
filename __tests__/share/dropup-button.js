import React from 'react';
import DropupBtn from '../../src/share/dropup-button';
import {mount} from 'enzyme';
import snapshot from '../snapshot';

describe('DropupBtn', () => {
  let props;
  beforeEach(() => {
    props = {
      btn: {
        disabled: false,
        kind: 'danger',
        btnText: 'link1'
      },
      links: [
        {
          onClick: () => console.log('clicked1'),
          separator: false,
          text: 'link 1',
        },
        {
          onClick: () => console.log('clicked2'),
          separator: false,
          text: 'link 2',
        }
      ]
    };
  });

  it('should render', () => {
    snapshot(<DropupBtn {...props} />);
  });

  it('should add open class when button is clicked', () => {
    const props = {
      btn: {
        disabled: false,
        kind: 'danger',
        btnText: 'My button',
      },
      links: [
        {
          onClick: () => console.log('clicked1'),
          separator: false,
          text: 'link 1',
        },
        {
          onClick: () => console.log('clicked2'),
          separator: false,
          text: 'link 2',
        }
      ],
    };
    const wrapper = mount(<DropupBtn {...props} />);
    const btn = wrapper.find('.btn');
    btn.simulate('click');
    expect(wrapper.find('.open').length).toBe(1);
  });
});
