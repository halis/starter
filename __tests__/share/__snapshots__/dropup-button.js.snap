exports[`DropupBtn should render 1`] = `
<div
  className="btn-group dropup dropup-button">
  <button
    className="btn btn-danger dropdown-toggle"
    disabled={false}
    onClick={[Function]}
    type="button">
    link1 
    <span
      className="caret" />
    <span
      className="sr-only">
      Toggle Dropdown
    </span>
  </button>
  <ul
    className="dropdown-menu">
    <li
      className={undefined}
      onClick={[Function]}
      role="">
      link 1
    </li>
    <li
      className={undefined}
      onClick={[Function]}
      role="">
      link 2
    </li>
  </ul>
</div>
`;
